### PR TITLE
Drulenium Visual Regression Quicksilver Integration

### PR DESCRIPTION
**Pull request by ndubbaka**
This is an example Quicksilver integration for Drulenium that will do an automatic visual regression test every-time you push code to development environment(s).
more at http://drulenium.org/how-to-automate-testing-in-pantheon-hosted-website
Demo video at http://drulenium.org/how-to-setup-drulenium-visual-testing-pantheon-quicksilver-platform-hook

### CODE REVIEW

File: drulenium_visual_testing/post_drulenium_github.php [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  'test_pages_sitemap' => array(
+    'boston',
+    'new-york',
+    'node',
+  ),
+);
+*/
+$test_definition = array (
+  'staging_environment_domain' => 'http://test-<add-my-site-name>.pantheonsite.io/',
+  'dev_environment_domain_suffix' => '-<add-my-site-name>.pantheonsite.io/',
```
Comment (ataylorme) [inline drulenium_visual_testing/post_drulenium_github.php]: @ndubbaka Have you considered replacing `<add-my-site-name>` with `$_ENV['PANTHEON_SITE_NAME']`? See our [doc on environment configuration](https://pantheon.io/docs/read-environment-config/) for available values

File: drulenium_visual_testing/post_drulenium_github.php [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    'add-my-third-page-here',
+  ),
+);
+/*----------END of REQUIRED Simple Configuration----------*/
+
+// Modify below variables if you are using your personal github account like https://github.com/Drulenium/pantheon-travis
+/*----------START of OPTIONAL Advanced Configuration-----------*/
+$github_username = 'Drulenium';
+$github_repository = 'pantheon-travis';
+$github_accesstoken = 'a06e6d536db8743056e1faae60aa803a0b17b13f';
```
Comment (ataylorme) [inline drulenium_visual_testing/post_drulenium_github.php]: @ndubbaka Github access token, email address and other items shouldn't be committed to version control. There can be a `private` directory on each Pantheon environment. It's best to create a `secrets.json` file in that directory and store private items there. See the [Slack integration example `README.md`](https://github.com/pantheon-systems/quicksilver-examples/blob/master/slack_notification/README.md) as an example.

**Comment by ndubbaka** [comment]
@ataylorme Thanks for your feedback. I addressed the issues you mentioned and updated the PR accordingly. Let me know how it looks.

File: drulenium_visual_testing/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+Setting up this example is easy:
+
+1. Add the example post_drulenium_github.php script to the 'private/scripts/' directory of your code repository.
+2. Add the secret github account parameters into a file called secrets.json and store it in the private files directory of Test & Live environments and Clone files from the test environment into other dev environments.
+
+  ```json
+    {
+      "github_username": "Drulenium",
+      "github_repository": "pantheon-travis",
+      "github_accesstoken": "ddedc2530c38734ea00651b80d41de5f1d2a64df",
```
Comment (ataylorme) [inline drulenium_visual_testing/README.md]: @ndubbaka I would remove your access token and put a placeholder here.

File: drulenium_visual_testing/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+This will allow you to do a comparative visual diff between the test environment and the development environemnt(including Multidev environments) everytime you push code to the development environment.
+
+As a note, Tests run on https://travis-ci.org/ and is free to use for unlimited number of pages but the test results are public. For more information read http://drulenium.org/how-to-automate-testing-in-pantheon-hosted-website
+
+## Instructions ##
+
+Setting up this example is easy:
+
+1. Add the example post_drulenium_github.php script to the 'private/scripts/' directory of your code repository.
+2. Add the secret github account parameters into a file called secrets.json and store it in the private files directory of Test & Live environments and Clone files from the test environment into other dev environments.
```
Comment (ataylorme) [inline drulenium_visual_testing/README.md]: @ndubbaka Can you specify the location of the private directory on the test/live environments? It should be clear to create `secrets.json` a user must connect to each environment via FTP and edit the file `files/private/secrets.json`.

I just don't want people to get the environment private directory confused with the private directory in the Git root.

**Comment by ataylorme** [comment]
> Thanks for your feedback. I addressed the issues you mentioned and updated the PR accordingly. Let me know how it looks.
@ndubbaka - no problem - thanks for submitting a PR! I added a few more comments on the update.

File: drulenium_visual_testing/README.md (hunk 3 above)
Comment (ndubbaka) [inline drulenium_visual_testing/README.md]: @ataylorme It's a test account and I am good with giving access to write into it for users who want to try this before setting up their own github account, travis account, Imgur.com account and configuring them accordingly.

This is the shared github account method of testing as described at http://drulenium.org/how-to-automate-testing-in-pantheon-hosted-website
image

**Comment by kyletaylored** [comment]
Drulenium no longer exists, closing this PR.

**Pull request closed without merging**